catalog/task/push-sbom-to-pyxis/0.1/push-sbom-to-pyxis.yaml

```diff
@@ -12,24 +12,90 @@ spec:
   description: >-
     Tekton task that extracts sbom from a pull spec and pushes it to Pyxis.
   params:
-    - name: imageURL
-      description: URL of the main binary image.
+    - name: imageURLs
+      description: Newline separated list of binary image URLs.
       type: string
+    - name: containerImageIDs
+      description: Newline separated list Pyxis image IDs
+      type: string
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'
+      default: production
   volumes:
     - name: workdir
       emptyDir: {}
   steps:
-    - name: download-sbom-for-image
-      # Image taken from https://github.com/sigstore/cosign
-      # and copied to Quay
+    - name: download-sbom-files
+      image:
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        IMAGEURLS=($(params.imageURLs))
+        IMAGEIDS=($(params.containerImageIDs))
+
+        if [ ${#IMAGEURLS[@]} != ${#IMAGEIDS[@]} ]; then
+          echo "ERROR: Lengths of imageURLs and containerImageIDs do not match"
+          exit 1
+        fi
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file ${IMAGEIDS[$i]}.json ${IMAGEURLS[$i]}
+        done
+
+    - name: push-sbom-files-to-pyxis
       image:
-        quay.io/redhat-appstudio/cosign:v1.13.1
-      args:
-        - download
-        - sbom
-        - --output-file
-        - /workdir/sbom-cyclonedx.json
-        - "$(params.imageURL)"
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
       volumeMounts:
         - mountPath: /workdir
           name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        if [[ "$(params.server)" == "production" ]]
+        then
+          PYXIS_URL="https://pyxis.api.redhat.com/"
+        elif [[ "$(params.server)" == "stage" ]]
+        then
+          PYXIS_URL="https://pyxis.preprod.api.redhat.com/"
+        else
+          echo "Invalid server parameter. Only 'production' and 'stage' are allowed."
+          exit 1
+        fi
+
+        echo "${pyxisCert}" > /workdir/crt
```

Comment:
Maybe we should reconsider the `set -x` that we have everywhere. In general it makes the log maybe too chatty. But also, in this case it will spit out your secrets to the log.

Reply:
good point

Reply:
Yeah there is no `set -x` in the `create-pyxis-image` task

Reply:
OK, I removed the `-x` here.

> Yeah there is no `set -x` in the `create-pyxis-image` task

Ah, you're right, I didn't notice it. But `create-pyxis-image` also doesn't have `set -e` which means that if one command in the script fails, the execution will carry on. E.g. if `skopeo inspect` fails, `create_container_image` will still be called. That doesn't seem right, WDYT @johnbieren ?

Reply:
https://github.com/hacbs-release/release-bundles/pull/93